lib/utf8.js

```diff
@@ -0,0 +1,12 @@
+"use strict";
+
+module.exports.decode = function (input) {
+  if (Buffer.isBuffer(input) === false) {
```

Comment:
That creates a problem for shims that are not based on `Buffer`. [utf8-typed](https://npmjs.com/utf8-typed), for example, would not need the `encode()`. Also, as an abstraction, having only "decode" still makes sense.